includes/class-sensei-lesson.php

```diff
@@ -4325,12 +4325,12 @@ public static function output_comments() {
 			return;
 		}
 
-		$allow_comments     = Sensei()->settings->settings['lesson_comments'];
-		$user_taking_course = Sensei_Utils::user_started_course( $course_id );
-		$has_access         = ! Sensei()->settings->get( 'access_permission' );
-		$is_preview         = Sensei_Utils::is_preview_lesson( $post->ID );
+		$allow_comments       = Sensei()->settings->settings['lesson_comments'];
+		$has_access           = ! Sensei()->settings->get( 'access_permission' );
+		$is_preview           = Sensei_Utils::is_preview_lesson( $post->ID );
+		$user_can_view_lesson = sensei_can_user_view_lesson();
 
-		$lesson_allow_comments = $allow_comments && ( $user_taking_course || $has_access || $is_preview );
+		$lesson_allow_comments = $allow_comments && ( $user_can_view_lesson || $has_access || $is_preview );
```

Comment:
Works well!

It seems though that since `sensei_can_user_view_lesson` already does the checks for `! Sensei()->settings->get( 'access_permission' )` and `Sensei_Utils::is_preview_lesson( $post->ID );`, we no longer need to duplicate those checks again here.

Reply:
Done!